lib/concurrent/edge/atomic_markable_reference.rb

```diff
@@ -0,0 +1,179 @@
+module Concurrent
+  module Edge
+    # @!macro atomic_markable_reference
+    class AtomicMarkableReference < ::Concurrent::Synchronization::Object
+      # @!macro [attach] atomic_markable_reference_method_initialize
+      def initialize(value = nil, mark = false)
+        super
```

Comment:
regarding the error, this should be `super()` so the args are not passed to parent initialize. I've mistakenly left old `initialize(*args, &block)` signature in ruby files (java extension has correct signature so it fails) so it was silently swallowing the args not using them.